app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,60 @@
+class CartsController < ApplicationController
+  before_action :init_cart
+  before_action :get_cart_book, except: :create
+
+  def new; end
+
+  def create
+    @cart_params = params[:cart]
+    @cart.each do |item|
+      @cur_cart_item = item if item["book_id"] == @cart_params["book_id"].to_i
```

Comment:
chỗ này ko cần biến instance đâu em, để biến local thôi hỉ 
```suggestion
      cur_cart_item = item if item["book_id"] == @cart_params["book_id"].to_i
```

Reply:
để local thế hàm if ở dưới k nhận đc ạ

Reply:
cur_cart_item = @cart.map do |item|
      next item if item["book_id"] == @cart_params["book_id"].to_i
      end
      Thử xem như này được không em

Reply:
map hay select ấy anh không nhớ rõ